httpdocs/ajax/server/session.php

```diff
@@ -16,13 +16,16 @@ function return_url($url) {
   error("can't connect to MySQL database: $mysqli->connect_error");
 $mysqli->set_charset('utf8');
 $url = $mysqli->escape_string($_GET['url']);
+$branch = basename(dirname(__FILE__, 4));
+
+$extra_condition = "branch=\"$branch\")";
 # search for a dedicated server first
-$query = "SELECT url FROM server WHERE `load` < 100 AND id IN (SELECT server FROM repository WHERE \"$url%\" LIKE CONCAT(url, '%')) ORDER BY `load` LIMIT 1";
+$query = "SELECT url FROM server JOIN server_branch ON server.id=server_branch.id WHERE `load` < 100 AND id IN (SELECT server FROM repository WHERE \"$url%\" LIKE CONCAT(url, '%')) AND $extra_condition ORDER BY `load` LIMIT 1";
 $result = $mysqli->query($query) or error($mysqli->error);
 if ($row = $result->fetch_array(MYSQLI_ASSOC))
   return_url($row['url']);
 # no available dedicated server found, reverting to a public server
-$query = "SELECT url FROM server WHERE `share` > 0 AND `share` - `load` > 0 ORDER BY `share` - `load` DESC LIMIT 1";
+$query = "SELECT url FROM server JOIN server_branch ON server.id=branch.id WHERE `share` > 0 AND `share` - `load` > 0 AND $extra_condition ORDER BY `share` - `load` DESC LIMIT 1";

```

Comment:
That's wrong:
```suggestion
$query = "SELECT url FROM server JOIN server_branch ON server.id=server_branch.id WHERE `share` > 0 AND `share` - `load` > 0 AND $extra_condition ORDER BY `share` - `load` DESC LIMIT 1";

```

It would be better to define:
```php
  $select = "SELECT url FROM server JOIN server_branch ON server.id=server_branch.id"
```
and re-use it instead of duplicating this code.